fpdf/output.py

```diff
@@ -1557,15 +1561,41 @@ def _finalize_catalog(
         catalog_obj.open_action = pdf_list(zoom_config)
         if struct_tree_root_obj:
             catalog_obj.mark_info = pdf_dict({"/Marked": "true"})
-        if fpdf.embedded_files:
-            file_spec_names = [
-                f"{PDFString(embedded_file.basename()).serialize()} {embedded_file.file_spec().serialize()}"
-                for embedded_file in fpdf.embedded_files
-                if embedded_file.globally_enclosed
-            ]
-            catalog_obj.names = pdf_dict(
-                {"/EmbeddedFiles": pdf_dict({"/Names": pdf_list(file_spec_names)})}
-            )
+        if fpdf.embedded_files or fpdf.named_destinations:
+            names_dict_entries = {}
+
+            if fpdf.embedded_files:
+                file_spec_names = [
+                    f"{PDFString(embedded_file.basename()).serialize()} {embedded_file.file_spec().serialize()}"
+                    for embedded_file in fpdf.embedded_files
+                    if embedded_file.globally_enclosed
+                ]
+                names_dict_entries["/EmbeddedFiles"] = pdf_dict(
+                    {"/Names": pdf_list(file_spec_names)}
+                )
+
+            if fpdf.named_destinations:
+                # Create a list of name/destination pairs for the Dests name tree
+                dests_names = []
+                for name, dest in fpdf.named_destinations.items():
+                    # Ensure the destination's page_ref is set
+                    if not hasattr(dest, "page_ref") or not dest.page_ref:
+                        page_index = dest.page_number - 1
+                        if 0 <= page_index < len(fpdf.pages):
+                            dest.page_ref = pdf_ref(fpdf.pages[dest.page_number].id)
+
+                    # Add name and destination to the Dests list
+                    dests_names.append(
+                        f"{PDFString(name).serialize()} {dest.serialize()}"
+                    )
+
+                if dests_names:
+                    names_dict_entries["/Dests"] = pdf_dict(
+                        {"/Names": pdf_list(dests_names)}
```

Comment:
```suggestion
                        {"/Names": pdf_list(sorted(dests_names))}
```

the /Names list must be in sorted order by name